OpenRA.Mods.D2k/Traits/SpiceBloom.cs

```diff
@@ -0,0 +1,168 @@
+		{
+			this.info = info;
+			self = init.Self;
+
+			resLayer = self.World.WorldActor.Trait<ResourceLayer>();
+			resType = self.World.WorldActor.TraitsImplementing<ResourceType>().First(t => t.Info.Name == info.ResourceType);
+
+			var render = self.Trait<RenderSprites>();
+			anim = new AnimationWithOffset(new Animation(init.Self.World, render.GetImage(self)), null, () => self.IsDead);
+			render.Add(anim);
+
+			respawnTicks = self.World.SharedRandom.Next(info.RespawnDelay[0], info.RespawnDelay[1]);
+			growTicks = self.World.SharedRandom.Next(info.GrowthDelay[0], info.GrowthDelay[1]);
+			anim.Animation.Play(info.GrowthSequences[0]);
+		}
+
+		public void Tick(Actor self)
+		{
+			if (!self.IsDead)
+			{
+				ticks++;
+
+				if (ticks >= growTicks)
+					self.Kill(self);
+				else
+				{
+					var index = info.GrowthSequences.Length * ticks / growTicks;
+					anim.Animation.Play(info.GrowthSequences[index]);
+				}
+			}
+		}
+
+		public void Killed(Actor self, AttackInfo e)
+		{
+			var weapon = self.World.Map.Rules.Weapons[info.Weapon.ToLowerInvariant()];
+
+			var pieces = self.World.SharedRandom.Next(info.Pieces[0], info.Pieces[1]) * ticks / growTicks;
+			for (var i = 0; pieces > i; i++)
+			{
+				var cells = OpenRA.Traits.Util.RandomWalk(self.Location, self.World.SharedRandom);
+				var cell = cells.Take(info.Range).SkipWhile(p => resLayer.GetResource(p) == resType && resLayer.IsFull(p)).Cast<CPos?>().RandomOrDefault(self.World.SharedRandom);
+				if (cell == null)
+					cell = cells.Take(info.Range).Random(self.World.SharedRandom);
+
+				var args = new ProjectileArgs
+				{
+					Weapon = weapon,
+					Facing = 0,
+
+					DamageModifiers = self.TraitsImplementing<IFirepowerModifier>()
+						.Select(a => a.GetFirepowerModifier()).ToArray(),
+
+					InaccuracyModifiers = self.TraitsImplementing<IInaccuracyModifier>()
+						.Select(a => a.GetInaccuracyModifier()).ToArray(),
+
+					Source = self.CenterPosition,
+					SourceActor = self,
+					PassiveTarget = self.World.Map.CenterOfCell(cell.Value)
+				};
+
+				self.World.AddFrameEndTask(_ =>
+				{
+					if (args.Weapon.Projectile != null)
+					{
+						var projectile = args.Weapon.Projectile.Create(args);
+						if (projectile != null)
+							self.World.Add(projectile);
+
+						if (args.Weapon.Report != null && args.Weapon.Report.Any())
+							Sound.Play(args.Weapon.Report.Random(self.World.SharedRandom), self.CenterPosition);
+					}
+				});
+			}
+
+			self.World.AddFrameEndTask(w => w.Add(new DelayedAction(respawnTicks, () =>
+			{
+				var td = new TypeDictionary
+				{
+					new ParentActorInit(self),
+					new LocationInit(self.Location),
+					new CenterPositionInit(self.CenterPosition),
+					new OwnerInit(self.Owner),
+					new FactionInit(self.Owner.Faction.InternalName),
+					new SkipMakeAnimsInit()
+				};
+				self.World.CreateActor(self.Info.Name, td);
+			})));
+		}
+
+		public void WarnCrush(Actor crusher) { }
```

Comment:
That didn't work out. As discussed in IRC that is quite infantry specific with `WarnCrush` overhead etc.